go/base/Dockerfile.tmpl

```diff
@@ -31,13 +31,18 @@ RUN apt-get -o Acquire::Check-Valid-Until=false update -y --no-install-recommend
 RUN ln -s /usr/bin/pip3 /usr/bin/pip
 {{- end }}
 
-COPY install-go.sh /tmp/install-go.sh
-RUN chmod ugo+rx /tmp/install-go.sh \
-  && /tmp/install-go.sh \
-  && rm /tmp/install-go.sh
+ARG GOLANG_VERSION=1.23.0
+ARG GOLANG_DOWNLOAD_URL=https://golang.org/dl/go$GOLANG_VERSION.linux-amd64.tar.gz
+ARG GOLANG_DOWNLOAD_SHA256=905a297f19ead44780548933e0ff1a1b86e8327bb459e92f9c0012569f76f5e3
+
+RUN curl -fsSL "$GOLANG_DOWNLOAD_URL" -o golang.tar.gz \
+	&& echo "$GOLANG_DOWNLOAD_SHA256  golang.tar.gz" | sha256sum -c - \
+        && rm -rf /usr/local/go \
+	&& tar -C /usr/local -xzf golang.tar.gz \
+	&& rm golang.tar.gz
```

Comment:
What's the reason for this change? If we do so, we need to accommodate the changes in the updatecli scripts too

Reply:
It was originally introduced in https://github.com/elastic/golang-crossbuild/pull/197

Reply:
I've just created https://github.com/elastic/golang-crossbuild/pull/440 to update how the golang installation is done. I initially wanted to reuse the former approach with the script... but I think I might keep this proposal and remove all the references to the script.